compiler/rustc_typeck/src/structured_errors/wrong_number_of_generic_args.rs

```diff
@@ -237,12 +678,91 @@ impl<'tcx> WrongNumberOfGenericArgs<'_, 'tcx> {
     /// type Map = HashMap<String, String, String, String>;
     /// ```
     fn suggest_removing_args_or_generics(&self, err: &mut DiagnosticBuilder<'_>) {
```

Comment:
To be honest, my eyes glazed over trying to read this method 😅

I wonder if we couldn't simplify a lot of the logic in this file by _always_ preparing a suggested string that contains all the correct generics, instead of trying to find the smallest possible span to give the suggestion.

Reply:
I agree that the method is still hard to read. I'm not sure I understand what you mean with your second sentence. Do you mean we should get rid of the span information for which arguments are redundant?

Reply:
I meant making the code _always_ synthesize `<'a, T>` regardless of whether just '`a`, just `T` or both `'a` and `T` are missing, and then the suggestion uses a single span to the whole generic parameter list. That way there's less logic as you don't need to tweak output based on what case you have.